Sessions Output

0Arian opened the pull request.

0Arian committed:
Updated the gen report to what I originally had in the other repo

0Arian committed:
Made the variable names more professional and commented out all the test echo's

0Arian committed:
Fixed arguments for API Key

0Arian committed:
Updated generate report to print out 7, 14, 21, 28, and 35 days of activity

DavidAwad commented:
@ArianMoslem start by following the PR template. 

Or at least something small like this: 

## What this is? 
this is a blah blah 

## How did I test this? 
make sure to `pip install <PACKAGE>`. Then run blah.py using a google analytics api key. 
here's a screenshot of the output and some of the command line output too. 


:)

davidawad commented on generate_report.py:
```diff
@@ -1,20 +1,151 @@
+      end_date='today',
+      metrics='ga:sessions').execute()
+
+def get_results14(service, profile_id):
```
make a more generic get_n_results funtion that takes n days instead of repeating like this.

davidawad commented on generate_report.py:
```diff
@@ -1,20 +1,151 @@
+  # Use the developer console and replace the values with your
+  # service account email and relative location of your key file.
+  service_account_email = 'analytics-sample@analytics-sample-168315.iam.gserviceaccount.com'
+  key_file_location = 'client_secrets.p12'
```
you might want to make these environment variables too.

davidawad reviewed (changes requested):
great start :+1:

davidawad commented on generate_report.py:
```diff
@@ -1,20 +1,151 @@
-# This file is part of ARM's analytics reporting which is released under the Apache 2 license.
-# See file /LICENSE or go to https://github.com/ARM-open/analytics-reports arm for full license details.
-# 
+"""A simple example of how to access the Google Analytics API."""
```
Add a shebang to specify what version of python this will be run with.

0Arian committed:
Fixed commenting changes and cleaned up get_results functions, then changed email and key file location to environment variables

0Arian committed:
Successful conversion from python2 into python3

0Arian committed:
type: feat

Imported PyGal and CairoSVG. With PyGal we are creating a bar chart and then rendering that chart into a svg file. With CairoSVG, we are taking the SVG file and pushing it into a pdf file.

0Arian committed:
type: feat - pdf arguments

Added argument parsing through click for PDF name.
